src/components/AgentMetrics.tsx

```diff
@@ -1,22 +1,126 @@
-import { Text, Flex, Grid, IconCaretDown, MenuProvider, MenuTrigger, MenuPopover, MenuSeparator, RadioGroup, Radio, Button, useRadioState } from "@vtex/shoreline";
+import { Button, Flex, Grid, IconCaretDown, MenuPopover, MenuProvider, MenuSeparator, MenuTrigger, Radio, RadioGroup, Skeleton, Text, useRadioState } from "@vtex/shoreline";
+import { useEffect, useMemo, useState } from "react";
+import { useSelector } from "react-redux";
+import { RootState } from "../interfaces/Store";
+import { agentCLI, agentMetrics, getSkillMetrics } from "../services/agent.service";
 import { DashboardItem } from "./DashboardItem";
-import { useEffect, useState } from "react";
-import { getSkillMetrics } from "../services/agent.service";
+
+function getPeriodDates(period: 'today' | 'yesterday' | 'last 7 days' | 'last 28 days') {
+  let startDate = '';
+  let endDate = '';
+
+  const toISODate = (date: Date) => date.toISOString().split('T')[0];
+
+  if (period === 'today') {
+    const today = new Date();
+
+    startDate = toISODate(today);
+    endDate = toISODate(today);
+  } else if (period === 'yesterday') {
+    const yesterday = new Date();
+
+    yesterday.setDate(yesterday.getDate() - 1);
+
+    startDate = toISODate(yesterday);
+    endDate = toISODate(yesterday);
+  } else if (period === 'last 7 days') {
+    const today = new Date();
+    const sevenDaysAgo = new Date();
+
+    sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);
+
+    startDate = toISODate(sevenDaysAgo);
+    endDate = toISODate(today);
+  } else if (period === 'last 28 days') {
+    const today = new Date();
+    const twentyEightDaysAgo = new Date();
+
+    twentyEightDaysAgo.setDate(twentyEightDaysAgo.getDate() - 28);
+
+    startDate = toISODate(twentyEightDaysAgo);
+    endDate = toISODate(today);
+  }
+
+  return { startDate, endDate };
+}
```

Comment:
I believe this code has room for improvement, especially in terms of reducing repetition, making it more extensible for future scenarios, and making it easier to write tests. Something like this, avoiding if-elses:

```suggestion
function getPeriodDates(period: 'today' | 'yesterday' | 'last 7 days' | 'last 28 days') {
  const toISODate = (date: Date) => date.toISOString().split('T')[0];

  const getStartAndEndDates = (daysAgo: number, includeToday: boolean = true) => {
    const end = new Date();
    const start = new Date();
    start.setDate(end.getDate() - daysAgo);
    return {
      startDate: toISODate(start),
      endDate: toISODate(includeToday ? end : start),
    };
  };

  const periodMap: Record<typeof period, () => { startDate: string; endDate: string }> = {
    'today': () => getStartAndEndDates(0),
    'yesterday': () => getStartAndEndDates(1, false),
    'last 7 days': () => getStartAndEndDates(7),
    'last 28 days': () => getStartAndEndDates(28),
  };

  return periodMap[period]();
}
```